eng/common/scripts/job-matrix/tests/job-matrix-functions.modification.tests.ps1

```diff
@@ -217,3 +217,47 @@ Describe "Platform Matrix Import" -Tag "import" {
         CompareMatrices $matrix $expected
     }
 }
+
+Describe "Platform Matrix Import/Replace" -Tag "replace" {
+    It "Should import a sparse matrix and replace values" {
+        $matrixJson = @'
+{
+    "matrix": {
+        "$IMPORT": "./test-import-matrix.json"
+    },
+    "replace": {
+        "Foo": {
+            "foo1": "foo1Replaced",
+            "foo.*": "fooDefaultReplaced"
+        },
+        ".*": {
```

Comment:
This is looking to be pretty close to the transform functionality you were thinking about adding. Do we think this is going to be the best level to do support this? I wonder if we should just do this similar to a matrix filter where we pass in a "replace" regex that can be used which we apply to all the names and/or values.